frontend/src/component/layout/MainLayout/DraftBanner/DraftBanner.tsx

```diff
@@ -37,43 +37,72 @@ const StyledBox = styled(Box)(({ theme }) => ({
                             <>
-                                in <strong>{changeRequest.environment} </strong>
+                                in{' '}
+                                {environments.map((env, i) =>
+                                    i === 0 ? (
+                                        <Fragment key={env}>
+                                            <strong>{env}</strong>
+                                        </Fragment>
+                                    ) : (
+                                        <Fragment key={env}>
+                                            {i === environments.length - 1
+                                                ? ' and '
+                                                : ', '}
+                                            <strong>{env}</strong>
+                                        </Fragment>
+                                    )
+                                )}
                             </>
                         }
                     />
                     <ConditionallyRender
-                        condition={changeRequest.state === 'Draft'}
-                        show={'that need to be reviewed'}
-                    />
-                    <ConditionallyRender
-                        condition={changeRequest.state === 'In review'}
-                        show={'that are in review'}
-                    />
-                    <ConditionallyRender
-                        condition={changeRequest.state === 'Approved'}
+                        condition={
+                            changeRequests.length === 1 &&
+                            ['Draft', 'In review', 'Approved'].includes(
+                                changeRequests[0].state
+                            )
+                        }
                         show={
-                            'that are approved. Adding more changes will clear the approvals and require a new review'
+                            {
+                                Draft: ' that need to be reviewed',
+                                'In review': ' that are in review',
+                                Approved:
+                                    ' that are approved. Adding more changes will clear the approvals and require a new review',
+                            }[
+                                changeRequests[0].state as
+                                    | 'Draft'
+                                    | 'In review'
+                                    | 'Approved'
+                            ]
                         }
                     />
+                    .
                 </Typography>
                 <Button
                     variant="contained"
                     onClick={onClick}
                     sx={{ ml: 'auto' }}
                 >
-                    View changes ({changesCount(changeRequest)})
+                    View changes (
+                    {changeRequests.reduce(
```

Comment:
I wouldn't put low level reduce inside the component. Scripting outside of the components with intention revealing names